process_data.go

```diff
@@ -206,22 +185,29 @@ func (dsp *DataStreamProcessor) AnalyzeData(records []*DataRecord) {
 		rec.pretrigMean = ptm
 		rec.peakValue = float64(max) - rec.pretrigMean
 
-		N := float64(dsp.NSamples - dsp.NPresamples)
+		N := float64(len(rec.data) - rec.presamples)
 		rec.pulseAverage = sum/N - ptm
 		meanSquare := sum2/N - 2*ptm*(sum/N) + ptm*ptm
 		rec.pulseRMS = math.Sqrt(meanSquare)
 		if !dsp.projectors.IsZero() {
 			rows, cols := dsp.projectors.Dims()
 			nbases := rows
-			if cols != len(rec.data) {
-				panic("wrong size")
-			}
-
+			// to support variable length records, we have to truncate projectors and basis
+			truncFront := dsp.NPresamples - rec.presamples
+			truncBack := dsp.NSamples - len(rec.data) - truncFront
+			i := 0
+			k := nbases
+			j := truncFront
+			l := cols - truncBack
+			projectors := dsp.projectors.Slice(i, k, j, l)
```

Comment:
Problem: the truncation of a basis does not produce projectors that are truncations of the originals. Our choices are:

1. Don't (yet) allow variable-length records. I know they will be important someday, but I am afraid that learning/deciding how to handle them might bog down this project considerably.

1. Have Dastard store projectors for all allowed lengths. (This might mean that we designate a set of allowed lengths, so that 500, 520, 540 are all allowed, while a record that you might want to be of length 534 has to be truncated to 520.)

1. Perform whitening or approximate whitening on the data in dastard. If you do this, then I think projection is still not as simple as truncating the projector, but it might be somewhat simpler.

Reply:
I thought you told me we could just truncate, but given that we can't, you're right so I took this out. I left in all the changes that make the summary quantities work with variable length records, and the tests on those quantities.